Well, time for you to follow approach 1 of removing numbers, 
- Write a function
- Implement the function on your column
- Vectorize your new column
- See how many columns you have. 

In [2]:
#Importing all the necessary libraries....
from nltk.corpus import stopwords
from nltk.tokenize import word_tokenize
import nltk
import pandas as pd
from nltk.stem import WordNetLemmatizer
lemmatizer = WordNetLemmatizer()
import nltk

# Main answer is in the last row of the notebook

# Prerequisite work 

In [3]:
#Reading the csv file in jupyter.
df=pd.read_csv('SPAM text message 20170820 - Data - SPAM text message 20170820 - Data.csv')

In [4]:
#To display the dataframe.
df

Category                                            Message
0         ham  Go until jurong point, crazy.. Available only ...
1         ham                      Ok lar... Joking wif u oni...
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...
3         ham  U dun say so early hor... U c already then say...
4         ham  Nah I don't think he goes to usf, he lives aro...
...       ...                                                ...
5567     spam  This is the 2nd time we have tried 2 contact u...
5568      ham               Will ü b going to esplanade fr home?
5569      ham  Pity, * was in mood for that. So...any other s...
5570      ham  The guy did some bitching but I acted like i'd...
5571      ham                         Rofl. Its true to its name

[5572 rows x 2 columns]

In [5]:
#Creating a function to remove all the stopwords in a sentence.
def datacleaning(x):
    y=nltk.word_tokenize(x)
    temp=[]
    for i in y:
      if i in stopwords.words("english"): 
         pass
      else:
         temp.append(i)
    my_new_string=' '.join(temp)
    return (my_new_string)

In [6]:
#Using the function on the column Message and storing the text in a new column called Clean_message.
df['Clean_message']=df['Message'].apply(datacleaning)

In [7]:
#display the dataframe
df

Category                                            Message  \
0         ham  Go until jurong point, crazy.. Available only ...   
1         ham                      Ok lar... Joking wif u oni...   
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...   
3         ham  U dun say so early hor... U c already then say...   
4         ham  Nah I don't think he goes to usf, he lives aro...   
...       ...                                                ...   
5567     spam  This is the 2nd time we have tried 2 contact u...   
5568      ham               Will ü b going to esplanade fr home?   
5569      ham  Pity, * was in mood for that. So...any other s...   
5570      ham  The guy did some bitching but I acted like i'd...   
5571      ham                         Rofl. Its true to its name   

                                          Clean_message  
0     Go jurong point , crazy .. Available bugis n g...  
1                       Ok lar ... Joking wif u oni ...  
2     Free entry 2 wkly comp win FA Cup final tkts 2...  
3           U dun say early hor ... U c already say ...  
4        Nah I n't think goes usf , lives around though  
...                                                 ...  
5567  This 2nd time tried 2 contact u. U £750 Pound ...  
5568                 Will ü b going esplanade fr home ?  
5569               Pity , * mood . So ... suggestions ?  
5570  The guy bitching I acted like 'd interested bu...  
5571                               Rofl . Its true name  

[5572 rows x 3 columns]

In [8]:
#Creating a function to lemmatize words in a sentence.
def my_lemmatizer(a):
    c=nltk.word_tokenize(a)
    my_list=[]
    for i in c:
        my_list.append(lemmatizer.lemmatize(i))
        my_new_string=' '.join(my_list)
    return my_new_string

In [9]:
#Using the function to lemmatize the text in every row of the clean_message and saving the lemmatized text in a ne column 
#called Lemmatized_message.
df['Lemmatized_message']=df['Clean_message'].apply(my_lemmatizer)

In [10]:
#Display the dataframe.
df

Category                                            Message  \
0         ham  Go until jurong point, crazy.. Available only ...   
1         ham                      Ok lar... Joking wif u oni...   
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...   
3         ham  U dun say so early hor... U c already then say...   
4         ham  Nah I don't think he goes to usf, he lives aro...   
...       ...                                                ...   
5567     spam  This is the 2nd time we have tried 2 contact u...   
5568      ham               Will ü b going to esplanade fr home?   
5569      ham  Pity, * was in mood for that. So...any other s...   
5570      ham  The guy did some bitching but I acted like i'd...   
5571      ham                         Rofl. Its true to its name   

                                          Clean_message  \
0     Go jurong point , crazy .. Available bugis n g...   
1                       Ok lar ... Joking wif u oni ...   
2     Free entry 2 wkly comp win FA Cup final tkts 2...   
3           U dun say early hor ... U c already say ...   
4        Nah I n't think goes usf , lives around though   
...                                                 ...   
5567  This 2nd time tried 2 contact u. U £750 Pound ...   
5568                 Will ü b going esplanade fr home ?   
5569               Pity , * mood . So ... suggestions ?   
5570  The guy bitching I acted like 'd interested bu...   
5571                               Rofl . Its true name   

                                     Lemmatized_message  
0     Go jurong point , crazy .. Available bugis n g...  
1                       Ok lar ... Joking wif u oni ...  
2     Free entry 2 wkly comp win FA Cup final tkts 2...  
3           U dun say early hor ... U c already say ...  
4           Nah I n't think go usf , life around though  
...                                                 ...  
5567  This 2nd time tried 2 contact u. U £750 Pound ...  
5568                 Will ü b going esplanade fr home ?  
5569                Pity , * mood . So ... suggestion ?  
5570  The guy bitching I acted like 'd interested bu...  
5571                               Rofl . Its true name  

[5572 rows x 4 columns]

In [11]:
#Creating a function to count the length of text.
def Text_count(x):
    y=nltk.word_tokenize(x)
    temp=[]
    for i in y: 
         temp.append(i)
    d=' '.join(temp)
    return(len(d.split()))

In [12]:
#counting the length of text in clean message using the function and saving it in a new colomn called length_text.
df['Lenghth_text']=df['Clean_message'].apply(Text_count)

In [13]:
#Display the dataframe
df

Category                                            Message  \
0         ham  Go until jurong point, crazy.. Available only ...   
1         ham                      Ok lar... Joking wif u oni...   
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...   
3         ham  U dun say so early hor... U c already then say...   
4         ham  Nah I don't think he goes to usf, he lives aro...   
...       ...                                                ...   
5567     spam  This is the 2nd time we have tried 2 contact u...   
5568      ham               Will ü b going to esplanade fr home?   
5569      ham  Pity, * was in mood for that. So...any other s...   
5570      ham  The guy did some bitching but I acted like i'd...   
5571      ham                         Rofl. Its true to its name   

                                          Clean_message  \
0     Go jurong point , crazy .. Available bugis n g...   
1                       Ok lar ... Joking wif u oni ...   
2     Free entry 2 wkly comp win FA Cup final tkts 2...   
3           U dun say early hor ... U c already say ...   
4        Nah I n't think goes usf , lives around though   
...                                                 ...   
5567  This 2nd time tried 2 contact u. U £750 Pound ...   
5568                 Will ü b going esplanade fr home ?   
5569               Pity , * mood . So ... suggestions ?   
5570  The guy bitching I acted like 'd interested bu...   
5571                               Rofl . Its true name   

                                     Lemmatized_message  Lenghth_text  
0     Go jurong point , crazy .. Available bugis n g...            20  
1                       Ok lar ... Joking wif u oni ...             8  
2     Free entry 2 wkly comp win FA Cup final tkts 2...            32  
3           U dun say early hor ... U c already say ...            11  
4           Nah I n't think go usf , life around though            10  
...                                                 ...           ...  
5567  This 2nd time tried 2 contact u. U £750 Pound ...            27  
5568                 Will ü b going esplanade fr home ?             8  
5569                Pity , * mood . So ... suggestion ?             9  
5570  The guy bitching I acted like 'd interested bu...            16  
5571                               Rofl . Its true name             5  

[5572 rows x 5 columns]

In [14]:
#Removing the numbers from the text.
def rem_num(x):
    temp=[]
    for i in x:
      if i.isdigit(): 
         pass
      else:
        temp.append(i)
    hi=''.join(temp)
    return hi

In [15]:
#Testing the function in a dummy text to see the impact.
rem_num('Hi i am 10 rasesh 20 and 30')

'Hi i am  rasesh  and '

In [16]:
#applying the function created to remove all the numbers form clean message to extract only the text and saving it in a new column
#called Rem_num.
df['Rem_num']=df['Clean_message'].apply(rem_num)

In [17]:
#Display the dataframe
df

Category                                            Message  \
0         ham  Go until jurong point, crazy.. Available only ...   
1         ham                      Ok lar... Joking wif u oni...   
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...   
3         ham  U dun say so early hor... U c already then say...   
4         ham  Nah I don't think he goes to usf, he lives aro...   
...       ...                                                ...   
5567     spam  This is the 2nd time we have tried 2 contact u...   
5568      ham               Will ü b going to esplanade fr home?   
5569      ham  Pity, * was in mood for that. So...any other s...   
5570      ham  The guy did some bitching but I acted like i'd...   
5571      ham                         Rofl. Its true to its name   

                                          Clean_message  \
0     Go jurong point , crazy .. Available bugis n g...   
1                       Ok lar ... Joking wif u oni ...   
2     Free entry 2 wkly comp win FA Cup final tkts 2...   
3           U dun say early hor ... U c already say ...   
4        Nah I n't think goes usf , lives around though   
...                                                 ...   
5567  This 2nd time tried 2 contact u. U £750 Pound ...   
5568                 Will ü b going esplanade fr home ?   
5569               Pity , * mood . So ... suggestions ?   
5570  The guy bitching I acted like 'd interested bu...   
5571                               Rofl . Its true name   

                                     Lemmatized_message  Lenghth_text  \
0     Go jurong point , crazy .. Available bugis n g...            20   
1                       Ok lar ... Joking wif u oni ...             8   
2     Free entry 2 wkly comp win FA Cup final tkts 2...            32   
3           U dun say early hor ... U c already say ...            11   
4           Nah I n't think go usf , life around though            10   
...                                                 ...           ...   
5567  This 2nd time tried 2 contact u. U £750 Pound ...            27   
5568                 Will ü b going esplanade fr home ?             8   
5569                Pity , * mood . So ... suggestion ?             9   
5570  The guy bitching I acted like 'd interested bu...            16   
5571                               Rofl . Its true name             5   

                                                Rem_num  
0     Go jurong point , crazy .. Available bugis n g...  
1                       Ok lar ... Joking wif u oni ...  
2     Free entry  wkly comp win FA Cup final tkts st...  
3           U dun say early hor ... U c already say ...  
4        Nah I n't think goes usf , lives around though  
...                                                 ...  
5567  This nd time tried  contact u. U £ Pound prize...  
5568                 Will ü b going esplanade fr home ?  
5569               Pity , * mood . So ... suggestions ?  
5570  The guy bitching I acted like 'd interested bu...  
5571                               Rofl . Its true name  

[5572 rows x 6 columns]

In [18]:
#After removing the numbers from clean message and saving it in rem_num column, we finally vectorize our column rem_num
from sklearn.feature_extraction.text import CountVectorizer
vectorizer=CountVectorizer()

In [19]:
#Fitting the text/string/column(Rem_num) in the vector.
vectorizer.fit(df['Rem_num'])

CountVectorizer()

In [20]:
#Getting the vocab out.
vectorizer.get_feature_names_out()

array(['____', 'aa', 'aah', ..., 'zyada', 'èn', '〨ud'], dtype=object)

In [21]:
#Getting the the first 100 vocabs out:
vectorizer.get_feature_names_out()[0:100]

array(['____', 'aa', 'aah', 'aaniye', 'aaooooright', 'aathi', 'ab',
       'abbey', 'abdomen', 'abeg', 'abel', 'aberdeen', 'abi', 'ability',
       'abiola', 'abj', 'able', 'abnormally', 'about', 'aboutas',
       'abroad', 'absence', 'absolutely', 'absolutly', 'abstract', 'abt',
       'abta', 'aburo', 'abuse', 'abusers', 'ac', 'academic', 'acc',
       'accent', 'accenture', 'accept', 'access', 'accessible',
       'accidant', 'accident', 'accidentally', 'accommodation',
       'accommodationvouchers', 'accomodate', 'accomodations', 'accordin',
       'accordingly', 'account', 'accounting', 'accounts', 'accumulation',
       'achan', 'ache', 'achieve', 'acid', 'acknowledgement', 'aclpm',
       'acnt', 'aco', 'across', 'act', 'acted', 'actin', 'acting',
       'action', 'activ', 'activate', 'active', 'activities', 'actor',
       'actual', 'actually', 'ad', 'ada', 'adam', 'add', 'addamsfa',
       'added', 'addicted', 'addie', 'adding', 'address', 'adds',
       'adewale', 'adi', 'ad

In [22]:
#Transforming the text/string/column(Rem_num) into vector
vector=vectorizer.transform(df['Rem_num'])

In [23]:
#To display the sparse matrix.
vector

<5572x7797 sparse matrix of type '<class 'numpy.int64'>'
	with 50587 stored elements in Compressed Sparse Row format>

In [24]:
#Converting the sparse matrix to array form and saving it in a a variable called x.
x=vector.toarray()

In [25]:
#Displaying the matrix
x

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

In [26]:
#Getting the number of columns/vocabs after removing the numbers.
len(vectorizer.get_feature_names_out())

7797

# The number of columns after removing the numbers are 7797.